library/kani_macros/src/lib.rs

```diff
@@ -39,37 +39,14 @@ pub fn proof(_attr: TokenStream, item: TokenStream) -> TokenStream {
     // )
 }
 
-#[cfg(kani)]
-#[proc_macro_attribute]
-pub fn proof(attr: TokenStream, item: TokenStream) -> TokenStream {
-    let mut result = TokenStream::new();
-
-    assert!(attr.is_empty(), "#[kani::proof] does not take any arguments");
-    result.extend("#[kanitool::proof]".parse::<TokenStream>().unwrap());
-    // no_mangle is a temporary hack to make the function "public" so it gets codegen'd
-    result.extend("#[no_mangle]".parse::<TokenStream>().unwrap());
-    result.extend(item);
-    result
-    // quote!(
-    //     #[kanitool::proof]
-    //     $item
-    // )
-}
-
-#[cfg(not(kani))]
-#[proc_macro_attribute]
-/// Treats #[kani::async_proof] like a normal #[kani::proof] if Kani is not active
-pub fn async_proof(attr: TokenStream, item: TokenStream) -> TokenStream {
-    proof(attr, item)
-}
-
-#[cfg(kani)]
-#[proc_macro_attribute]
-/// Translates #[kani::async_proof] to a #[kani::proof] harness that calls `kani::block_on`, if Kani is active
+/// Handles #[kani::proof] if #[cfg(kani)] is active
+///
+/// For synchronous functions, it becomes a Kani-internal attribute.
```

Comment:
Usually proc macros don't describe their implementation details. It's better to spend the time to talk about how to use it and what it's for.

Reply:
Good point. I will update the comment.

Reply:
I've updated the doc comment accordingly. If there's anything else you think it should include from a user's perspective, please let me know.